federation/manifests/federation-controller-manager-deployment.yaml

```diff
@@ -17,16 +17,23 @@ spec:
       - name: ssl-certs
         hostPath:
           path: /etc/ssl/certs
+      - name: kubeconfig
+        secret:
+          secretName: federation-apiserver-kubeconfig
```

Comment:
Actually I am not sure if we will fallback to DeprecatedKubeconfigSecretName if value of kubeconfig flag is empty. Will  BuildConfigFromFlags generate an error if kubeconfig is empty string?